### PR TITLE
swarm/network: update syncer metrics

### PR DESCRIPTION
**Pull request by acud**

### CODE REVIEW

File: swarm/network/stream/syncer.go [hunk 1]
```diff
@@ -111,13 +111,13 @@ func (s *SwarmSyncerServer) SetNextBatch(from, to uint64) ([]byte, uint64, uint6
 		timerC       <-chan time.Time
 	)
 
-	defer func() {
-		metrics.GetOrRegisterResettingTimer("syncer.set-next-batch.total-time", nil).UpdateSince(batchStart)
-		metrics.GetOrRegisterCounter("syncer.set-next-batch.batch-size", nil).Inc(int64(batchSize))
+	defer func(start time.Time) {
+		metrics.GetOrRegisterResettingTimer("syncer.set-next-batch.total-time", nil).UpdateSince(start)
+		metrics.GetOrRegisterGauge("syncer.set-next-batch.batch-size", nil).Update(int64(batchSize))
```
Comment (nonsense) [inline swarm/network/stream/syncer.go]: Not sure about that - you have many syncers in a given process, so having a counter gives you the opportunity to count all batch sizes and sum them up.

Having this as a Gauge, which all syncers will modify doesn't make much sense to me. A Gauge is a singleton so to say.

Comment (acud) [reply swarm/network/stream/syncer.go]: But the counter doesn't allow me to see the individual increments, i.e. I can't see the individual batch size

Comment (nonsense) [reply swarm/network/stream/syncer.go]: A Gauge won't allow you to see the individual changes either.

Comment (acud) [reply swarm/network/stream/syncer.go]: AFAIK Gauge will, but if I `sum` the gauge readings I'll get to the same measurement as the `Counter`s

Comment (nonsense) [reply swarm/network/stream/syncer.go]: Gauge is losing the updates you make during the resolution time (which is 10sec.). It does not keep a history of all the changes, and it seems like this is what you need.

Comment (acud) [reply swarm/network/stream/syncer.go]: done :+1:

**Pull request merged**